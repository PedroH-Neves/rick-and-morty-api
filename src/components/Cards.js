import { useSelector } from 'react-redux';
import '../styles/components/Cards.scss';
import { Link } from 'react-router-dom';
import Char from './Char';

const Cards = () => {
  const charactersState = useSelector((state) => state.charactersReducers.characters);
  const filterState = useSelector((state) => state.filter);

  const renderChars = (
    filterState.length === 0
      ? charactersState.map((char) => (
        <Link to={`/details/${char.id}`} className="characters-card" key={char.name} id={char.id}>
          <Char name={char.name} image={char.image} />
        </Link>
      ))
      : filterState[0].map((char) => (
        <Link to={`/details/${char.id}`} className="characters-card" key={char.name} id={char.id}>
          <Char name={char.name} image={char.image} />
        </Link>
      ))
  );

  return (
    <div className="cards-list">
      {renderChars}
    </div>
  );
};

export default Cards;
