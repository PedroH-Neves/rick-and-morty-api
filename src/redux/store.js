import { configureStore } from "@reduxjs/toolkit";
import charactersSlice from "./characters/charactersSlice";

const store = configureStore({
  reducer: {
    characters: charactersSlice,
  }
}) 

export default store;