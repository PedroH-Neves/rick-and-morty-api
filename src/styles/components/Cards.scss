.cards-list {
  margin: 0 auto;
  background: #ec4c8a;
  padding: 3rem 10rem;
  display: grid;
  grid-template-columns: repeat(3, minmax(200px, 1fr));
  text-decoration: none;

  @media (max-width: 992px) {
    padding: 2.5rem 3rem;
    grid-template-columns: repeat(2, minmax(200px, 2fr));
  }

  @media (max-width: 580px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
}