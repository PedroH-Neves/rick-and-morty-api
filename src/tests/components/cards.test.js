import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Cards from '../../components/Cards';
import store from '../../redux/store';

const character = {
  name: 'Rick Sanchez',
  image: 'character image',
};

it('render all characters', () => {
  const characters = render;

  render(
    <Provider store={store}>
      <Cards name={character.name} image={character.image} />
    </Provider>,
  );

  expect(characters).toMatchSnapshot();
});
