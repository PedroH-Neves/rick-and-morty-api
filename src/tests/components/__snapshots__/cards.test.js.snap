// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all characters 1`] = `[Function]`;
